examples/client.rs

```diff
@@ -0,0 +1,39 @@
+use imap_codec::imap_types::{
+    command::{Command, CommandBody},
+    core::Tag,
+};
+use imap_flow::{
+    client::{ClientFlow, ClientFlowEvent, ClientFlowOptions},
+    stream::AnyStream,
+};
+use tokio::net::TcpStream;
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() {
+    let stream = TcpStream::connect("127.0.0.1:12345").await.unwrap();
+
+    let (mut client, greeting) =
+        ClientFlow::receive_greeting(AnyStream::new(stream), ClientFlowOptions::default())
+            .await
+            .unwrap();
+    println!("received greeting: {greeting:?}");
+
+    client.enqueue_command(Command {
+        tag: Tag::try_from("A1").unwrap(),
+        body: CommandBody::Noop,
+    });
+
+    loop {
+        match client.progress().await.unwrap() {
+            ClientFlowEvent::CommandSent { tag, handle } => {
```

Comment:
It's a bit weird to have `tag` and `handle` here. Ideally, the handle can be used to obtain the command. And when this can be done, the `tag` can be obtained through the obtained command.

I stumbled upon this because I wanted to `Debug` print the sent command. So, I needed to use an external `HashMap<Handle, Command>`. But after `let cmd = map.remove(handle)`, I can already access the tag through `cmd`.

Maybe there should be a `handle.into_command()` or so?

Reply:
Yeah. It's also not entirely clear to me how to make good use of the handle.

The loop could, in theorie, yield one or more received events before the sent event. Are we expected to take this into account? E.g., to diverge our logic into pre- and post arrival of the sent event?

Are we expected to verify that the thing we `enqued` was really sent? Or do we implicitly assume that our command was processed because we got a corresponding command completion result (status with same tag as command)?

Reply:
These were the questions I had when implementing a draft version of `client.login()`, `client.fetch()`, etc.

Spoiler: It all works amazingly well so far. But I mostly just ignore `SentCommand`.

Reply:
Resolving for now. Will open an issue.